File: VB/DiagramDataControllerBehavior/Form1.vb
Imports DiagramDataControllerBehavior.Data
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace DiagramDataControllerBehavior
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm

		Public Sub New()
			InitializeComponent()
			diagramDataBindingController1.DataSource = ClassStructureGenerator.ClassList()
			diagramDataBindingController1.ConnectorsSource = ClassStructureGenerator.ConnectionList()

		End Sub

		Private Sub diagramDataBindingController1_GenerateConnector(ByVal sender As Object, ByVal e As DevExpress.XtraDiagram.DiagramGenerateConnectorEventArgs) Handles diagramDataBindingController1.GenerateConnector
			If CType(e.From, ClassData).Type = ClassType.Interface OrElse CType(e.To, ClassData).Type = ClassType.Interface Then
				e.Connector = e.CreateConnectorFromTemplate("InterfaceConnector")
			Else
				e.Connector = e.CreateConnectorFromTemplate("ClassConnector")
			End If
		End Sub

		Private Sub diagramDataBindingController1_GenerateItem(ByVal sender As Object, ByVal e As DevExpress.XtraDiagram.DiagramGenerateItemEventArgs) Handles diagramDataBindingController1.GenerateItem
			If CType(e.DataObject, ClassData).Type = ClassType.Interface Then
				e.Item = e.CreateItemFromTemplate("InterfaceShape")
			Else
				e.Item = e.CreateItemFromTemplate("ClassShape")
			End If
		End Sub
	End Class
End Namespace

File: CS/DiagramDataControllerBehavior/Form1.cs
using DiagramDataControllerBehavior.Data;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DiagramDataControllerBehavior
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            diagramDataBindingController1.DataSource = ClassStructureGenerator.ClassList();
            diagramDataBindingController1.ConnectorsSource = ClassStructureGenerator.ConnectionList();
           
        }

        private void diagramDataBindingController1_GenerateConnector(object sender, DevExpress.XtraDiagram.DiagramGenerateConnectorEventArgs e)
        {
            if (((ClassData)e.From).Type == ClassType.Interface || ((ClassData)e.To).Type == ClassType.Interface)
                e.Connector = e.CreateConnectorFromTemplate("InterfaceConnector");
            else e.Connector = e.CreateConnectorFromTemplate("ClassConnector");
        }

        private void diagramDataBindingController1_GenerateItem(object sender, DevExpress.XtraDiagram.DiagramGenerateItemEventArgs e)
        {
            if (((ClassData)e.DataObject).Type == ClassType.Interface)
                e.Item = e.CreateItemFromTemplate("InterfaceShape");
            else e.Item = e.CreateItemFromTemplate("ClassShape");
        }
    }
}
